plasmapy/dispersion/numerical/hollweg_.py

```diff
@@ -0,0 +1,321 @@
+            )
+    if not (ion.is_ion or ion.is_category("element")):
+        raise ValueError("The particle passed for 'ion' must be an ion or element.")
+
+    # validate z_mean
+    if z_mean is None:
+        try:
+            z_mean = abs(ion.integer_charge)
+        except ChargeError:
+            z_mean = 1
+    else:
+        if not isinstance(z_mean, (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument 'z_mean', but got {type(z_mean)}."
+            )
+        z_mean = abs(z_mean)
+
+    # validate arguments
+    for arg_name in ("B", "n_i", "T_e", "T_i"):
+        val = locals()[arg_name].squeeze()
+        if val.shape != ():
+            raise ValueError(
+                f"Argument '{arg_name}' must a single value and not an array of "
+                f"shape {val.shape}."
+            )
+        locals()[arg_name] = val
+
+    # validate arguments
+    for arg_name in ("gamma_e", "gamma_i"):
+        if not isinstance(locals()[arg_name], (int, np.integer, float, np.floating)):
+            raise TypeError(
+                f"Expected int or float for argument '{arg_name}', but got "
+                f"{type(locals()[arg_name])}."
+            )
+
+    # validate argument k
+    k = k.squeeze()
+    if not (k.ndim == 0 or k.ndim == 1):
+        raise ValueError(
+            f"Argument 'k' needs to be a single valued or 1D array astropy Quantity,"
+            f" got array of shape {k.shape}."
+        )
+    if np.any(k <= 0):
+        raise ValueError("Argument 'k' can not be a or have negative values.")
+
+    # validate argument theta
+    theta = theta.squeeze()
+    theta = theta.to(u.radian)
+    if not (theta.ndim == 0 or theta.ndim == 1):
+        raise ValueError(
+            f"Argument 'theta' needs to be a single valued or 1D array astropy "
+            f"Quantity, got array of shape {k.shape}."
+        )
+
+    # Calc needed plasma parameters
+    n_e = z_mean * n_i
+    c_s = pfp.ion_sound_speed(
+        T_e=T_e,
+        T_i=T_i,
+        ion=ion,
+        n_e=n_e,
+        gamma_e=gamma_e,
+        gamma_i=gamma_i,
+        z_mean=z_mean,
+    )
+    v_A = pfp.Alfven_speed(B, n_i, ion=ion, z_mean=z_mean)
+    omega_ci = pfp.gyrofrequency(B=B, particle=ion, signed=False, Z=z_mean)
+    omega_pe = pfp.plasma_frequency(n=n_e, particle="e-")
+
+    # Parameters kx and kz
+    kz = np.cos(theta.value) * k
+    kx = np.sqrt(k ** 2 - kz ** 2)
+
+    # Bellan2012JGR beta param equation 3
+    beta = (c_s / v_A) ** 2
+
+    # Parameters D, F, sigma, and alpha to simplify equation 3
+    D = (c_s / omega_ci) ** 2
+    F = (c / omega_pe) ** 2
+    sigma = (kz * v_A) ** 2
+    alpha = (k * v_A) ** 2
+
+    # Polynomial coefficients: c3*x^3 + c2*x^2 + c1*x + c0 = 0
+    c3 = (F * kx ** 2 + 1) / sigma
+    c2 = -((alpha / sigma) * (1 + beta + F * kx ** 2) + D * kx ** 2 + 1)
+    c1 = alpha * (1 + 2 * beta + D * kx ** 2)
+    c0 = -beta * alpha * sigma
```

Comment:
I went through the math and was able to derive the same expressions for these coefficients.  However, this formulation will not play nicely with the edge case of `kz == 0` (i.e. `sigma == 0`).  Not sure what the best approach is to remedy this, but most likely just multiple the expressions by `sigma` to eliminate the divide by zero.

Reply:
I forgot to consider the divide by 0 case. I multiplied everything by sigma and double checked that the output was the same just be thorough. This should be fixed now.